pkg/auth/authenticator/password/basicauthpassword/basicauthpassword.go

```diff
@@ -49,8 +50,12 @@ type RemoteError struct {
 }
 
 // New returns an authenticator which will make a basic auth call to the given url.
-func New(providerName string, url string, mapper authapi.UserIdentityMapper) authenticator.Password {
-	return &Authenticator{providerName, url, mapper}
+func New(providerName string, url string, transport http.RoundTripper, mapper authapi.UserIdentityMapper) authenticator.Password {
```

Comment:
Improve the comment to describe what's done with the round tripper, why I'd want to specify one and a reference to the method that can build me one.